pyvelox/pyvelox.h

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+#include <pybind11/stl_bind.h>
+#include <velox/type/Type.h>
+
+namespace facebook::pyvelox {
+
+std::string serializeType(const std::shared_ptr<const velox::Type>& type);
+
+// Inlining these bindings since adding them to the cpp file results in a
+// ASAN error.
+inline void addVeloxBindings(pybind11::module& m, bool asLocalModule = true) {
+  using namespace velox;
+  namespace py = pybind11;
+
+  py::enum_<velox::TypeKind>(m, "TypeKind", py::module_local(asLocalModule))
+      .value("BOOLEAN", velox::TypeKind::BOOLEAN)
+      .value("TINYINT", velox::TypeKind::TINYINT)
+      .value("SMALLINT", velox::TypeKind::SMALLINT)
+      .value("INTEGER", velox::TypeKind::INTEGER)
+      .value("BIGINT", velox::TypeKind::BIGINT)
+      .value("REAL", velox::TypeKind::REAL)
+      .value("DOUBLE", velox::TypeKind::DOUBLE)
+      .value("VARCHAR", velox::TypeKind::VARCHAR)
+      .value("VARBINARY", velox::TypeKind::VARBINARY)
+      .value("TIMESTAMP", velox::TypeKind::TIMESTAMP)
+      .value("OPAQUE", velox::TypeKind::OPAQUE)
+      .value("ARRAY", velox::TypeKind::ARRAY)
+      .value("MAP", velox::TypeKind::MAP)
+      .value("ROW", velox::TypeKind::ROW)
+      .export_values();
+
+  py::class_<Type, std::shared_ptr<Type>> type(
+      m, "VeloxType", py::module_local(asLocalModule));
+
+  py::class_<BooleanType, Type, std::shared_ptr<BooleanType>> booleanType(
```

Comment:
as a general comment, it would be nice to comment this header with what each of these declaration do, what they export to python, and how they are accessed.